BaseClasses.py

```diff
@@ -300,6 +300,51 @@ def link_items(self) -> None:
         """Called to link together items in the itempool related to the registered item link groups."""
         from worlds import AutoWorld
 
+        def patch_collect_and_remove(group_world: AutoWorld.World, linked_items: Dict[str, List[List[Item]]]):
```

Comment:
I could not find a good place to modify the collect/remove methods. I don't think monkey patching is great, but I don't think there are any good options.

`MultiWorld.add_group()` currently contains no behaviour that is clearly specific for supporting item links. From what I've seen discussed, the less that core has to specifically support item links behaviour the better.

Even if the collect/remove methods for handling item links were set in `MultiWorld.add_group()`, the item links have not been set up at that point, meaning that the collect/remove methods would have to get the linked `Item`s to collect/remove from somewhere, likely the group's `World` or the `Group` (`TypedDict`) instance itself, and `MultiWorld.link_items()` would have to set the linked `Item`s onto whichever type is chosen.
- Storing the linked `Item`s on the `Group` isn't ideal because the `Group` already contains a reference to the `World` and this would result in the `World`'s collect/remove now having reference to the `Group`. At which point, it feels like they should just be stored on the `World` instead, however...
- Storing the linked `Item`s on the `World` isn't ideal because to get any type checking for where the linked `Item`s are stored would requiring modifying `AutoWorld.World` to include a typed attribute for item links. As has already been mentioned, it is preferable to avoid modifying core specifically to support item links.
- Possibly the linked `Item`s could be stored on the `MultiWorld` itself since `MultiWorld` already contains some the `item_links` attribute that is specific to supporting item links, though this would require an extra step of storing all the linked `Item`s in a dictionary keyed by the group ID because there is only one `MultiWorld` shared by all the `World`s/`Group`s.